sentinel/src/transitions/accept_patient_reports.js

```diff
@@ -70,16 +70,14 @@ const findToClear = (registration, reported_date, config) => {
     silenceUntil.add(date.getDuration(config.silence_for));
 
     const allTasksBeforeSilenceUntil = tasksUnderReview.filter(
-      task => moment(task.due) <= silenceUntil
+      task => moment(task.due) <= silenceUntil && statesToClear.includes(task.state)
```

Comment:
I don't think this is correct. I recently (last week) went through this part of the code and thought about this logic. 

Imagine this:
You have a `pregnancy` form which adds a `Pregnancy Visit Schedule`. 
The schedule requires 3 prenatal visit from the CHW at 13, 25 and 36 weeks, so 3 groups.
Each of the groups has 3 messages of type `Pregnancy visit reminder`, one week before the task due date, the week of the due date and one week after the due date.

You also have a `patient_reports` setting for `pregnancy_visit` form, which should clear these schedules. 

So, say you are are at the end of week 12 and you've received the 1st reminder to visit next week.
You submit a `pregnancy_visit` form. This submission should clear all your group (1) tasks.

So, your tasks will be:
{ week: 12, status: sent }, { week: 13, status: scheduled }, { week: 14, status: scheduled }.

If we were to include this change, then the week 12 status sent will not be filtered, causing the whole group to remain scheduled, the CHW receiving 2 more reminders for a task that is already done. 

If you want a concrete example, `standard` config implements something similar with ANC schedules. 

I think we should leave it as is, allowing sent tasks to clear later scheduled tasks of the same type and part of the same group.

Reply:
👍